### PR TITLE
Participant Visit Listing v1.3.1

### LINKED ISSUE #87: Calculate number of overdue visits with analysis subset.
still participant-level, but calculate after applying the analysis subsets

# Test notes
* The options in the _# of Overdue Visits_ dropdown should update after changing an analysis subset filter.
  1. Set _Analysis Subset 2_ to _On Treatment_.
  2. Verify that the options update in the _# of Overdue Visits_ dropdown.
* The selected option in the _# of Overdue Visits_ dropdown should update to _All_ if the selected option does not exist in analysis subset.
  1. Set the _# of Overdue Visits_ dropdown to _2_.
  2. Set _Analysis Subset 2_ to _On Treatment_.
  3. Verify that the selected option in the _# of Overdue Visits_ dropdown updates to _All_.

### LINKED ISSUE #88: Set resizable to false.
resizing a huge chart is time-consuming

# Test notes
* Change the zoom of the page and verify the chart does not resize.

### LINKED ISSUE #90: Add a destroy method.
That not only clears the contents of the containing element and the Webcharts object(s), but the stylesheet as well.

# Test notes
* Open the developer console.
*  On the Elements tab expand the `<head>` element and verify that the last child node is a `<style>` element and that its styles include _.participant-visit-listing_ classes.
* On the Console tab run `d3.select('.participant-visit-listing').node()` and verify the Participant Visit Listing `<div>` element is returned.
* Run `d3.select('.participant-visit-listing').datum().destroy()` and verify both the `<style>` element and `<div>` element are removed from the DOM.

Reference the images in [this crf-heat-map issue](https://github.com/RhoInc/crf-heat-map/issues/135) for screenshots.

### PR DESCRIPTION
**Pull request by samussiah**
# Issues
Closes #90 
Closes #88 
Closes #87

### CODE REVIEW

File: scripts/save-data.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+        // A chunk of data has been recieved.
+        resp.on('data', (chunk) => {
+            csv += chunk;
+        });
+
+        // The whole response has been received. Print out the result.
+        resp.on('end', () => {
+            const json = d3.csv.parse(csv);
+            fs.writeFileSync('./test/visits.json', JSON.stringify(json, null, 4), 'utf8');
```
Comment (pburnsdata) [inline scripts/save-data.js]: easter egg!

File: src/destroy.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { select } from 'd3';
+
+export default function destroy() {
+    //Remove displays.
+    this.ordinalChart.destroy();
+    this.linearChart.destroy();
+    this.listing.destroy();
+
+    //Remove stylesheet.
+    this.style.remove();
```
Comment (pburnsdata) [inline src/destroy.js]: good call on attaching the styling to the object - makes this way cleaner and safer

File: src/charts/functions/onLayout/addButtons/split.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 export default function split() {
-    const t0 = performance.now();
+    const t0 = this.pvl.performance.now();
```
Comment (pburnsdata) [inline src/charts/functions/onLayout/addButtons/split.js]: i noticed these performance.nows() are causing errors in cat if you try and change the tab - but i feel like its on the overall object - maybe its just the wrong this here?

File: scripts/save-data.js (hunk 1 above)
Comment (samussiah) [reply scripts/save-data.js]: Unit tests need data too!

File: src/charts/functions/onLayout/addButtons/split.js (hunk 3 above)
Comment (samussiah) [reply src/charts/functions/onLayout/addButtons/split.js]: Good catch.  Got the context mixed up in addtabFunctionality.

File: scripts/save-data.js (hunk 1 above)
Comment (pburnsdata) [reply scripts/save-data.js]: those needy unit tests!

File: test/participantVisitListing.js [hunk 4]
```diff
@@ -37,7 +46,7 @@ describe('The participantVistiListing function is called.', () => {
             'tabs',
             'ordinalChart',
             'linearChart',
-            'listing'
+            'listing',
```
Comment (pburnsdata) [inline test/participantVisitListing.js]: @samussiah do we want this comma?

Comment (samussiah) [reply test/participantVisitListing.js]: Not really but it won't hurt anything.

**Pull request merged**